formam.go

```diff
@@ -295,19 +295,10 @@ func (dec *Decoder) traverse() error {
 			dec.traverseInMap(true)
 		}
 		dec.field = ""
+
+		dec.traverseIndirect()
```

Comment:
I believe it must be nested in both cases `dec.field != ""` and `dec.index != ""`  
otherwise calls to `dec.traverse()` with empty `field` and `index` changes decoder state if `dec.curr` points to some `reflect.Pointer`